### ISSUE TITLE
Support Thumbnails in KDE (Dolphin)

### ISSUE THREAD

**Issue by geowarin** [f3d-app/f3d#1268]
**Is your feature request related to a problem? Please describe.**

F3D gorgeous thumbnails do not show on KDE.
KDE appears to have a custom protocol (ie does not support standard thumbnailers).

See: https://discuss.kde.org/t/3d-file-thumbnailer/4173

**Describe the solution you'd like**

Write a plugin to supports thumbnails in KDE.

After a quick inspection, a simple KDE thumbnailer is not too complex.

For instance the [blend file thumbnailer](https://invent.kde.org/graphics/kdegraphics-thumbnailers/-/tree/master/blend?ref_type=heads) is composed of 3 files.

**Describe alternatives you've considered**

Using another file manager :cry: 

**Contribution**

I have no experience in KDE development but if F3D is interested in supporting this features I could probably lend a hand!

**Comment by Meakk** [comment]
That looks surprisingly simple. We should be able to rely on `f3d::window::renderToImage` and convert the result to the `QByteArray`.

**Comment by mwestphal** [comment]
That'd be nice, but can we ship it ?

In any case, we can create this plugin In Another repo and rely on packager to ship it.

We can even put it in F3D but not enable it in the binary release.

**Comment by Meakk** [comment]
For reference, Arch package containing the thumbnailers : https://archlinux.org/packages/extra/x86_64/kdegraphics-thumbnailers/  
There is a dependency on `ghostscript` for example to generate thumbnails for `.ps` files.  
Should we contribute directly there?

**Comment by mwestphal** [comment]
Hum, that is an interesting approach. here is the source code:
 https://invent.kde.org/graphics/kdegraphics-thumbnailers
 
 We would need to discuss with the maintainers of this project

**Comment by mwestphal** [comment]
https://www.linkedin.com/in/nicolas-fella-460565161/

https://www.linkedin.com/in/lmontel/

**Comment by mwestphal** [comment]
No success via LinkedIn, I sent a mail.

**Comment by adro79** [comment]
> No success via LinkedIn, I sent a mail.

Any updates on this?

**Comment by mwestphal** [comment]
I'm afraid I heard nothing back at all.

**Comment by mwestphal** [comment]
Finally got a clear answer from a dev here: https://invent.kde.org/graphics/kdegraphics-thumbnailers/-/issues/1

TLDR: KDE will move to standard thumbnailer mechanism, as all other file managers.

**Comment by Doroszewski** [comment]
The mechanism has been implemented and is supposed to arrive with KDE Frameworks 6.6 in September.[^1][^2]

[^1]: https://pointieststick.com/2024/08/09/this-week-in-kde-svg-breeze-cursors-and-more-thumbnails/
[^2]: https://akselmo.dev/posts/kio-thumbnailer-support/

**Comment by mwestphal** [comment]
nice! Then installing F3D should be enough to get thumbnails with KDE!
